spike_main/spike.cc

```diff
@@ -299,10 +299,28 @@ static std::vector<int> parse_hartids(const char *s)
 
   int n;
   while (stream >> n) {
+    if (n < 0) {
```

Comment:
Using `uint32_t` would ignore a negative number rather than rejecting it.  But we can do what you suggested and improve robustness further by checking the stream is empty after the loop.

Reply:
There's also a mix of `size_t` and `uint32_t` elsewhere used for hartids. I don't know if that's anything to worry about.

Reply:
Turns out `stream >> unsigned` has unintuitive behavior (it happily accepts negative integers takes them mod 2^width) and so the existing code is actually reasonable.  I'm going to mark this as resolved.  (I'm still going to handle the `size_t` vs. `uint32_t` vs. `int` matter elsewhere, though.)